Neos.ContentRepository/Classes/Domain/Model/NodeType.php

```diff
@@ -175,23 +175,12 @@ protected function buildFullConfiguration()
      *
      * @return array
      */
-    protected function buildInheritanceChain()
+    public function buildInheritanceChain()
     {
-        $superTypes = array();
-        foreach ($this->declaredSuperTypes as $superTypeName => $superType) {
-            if ($superType !== null) {
-                $this->addInheritedSuperTypes($superTypes, $superType);
-                $superTypes[$superTypeName] = $superType;
-            }
-        }
-
-        foreach ($this->declaredSuperTypes as $superTypeName => $superType) {
-            if ($superType === null) {
-                unset($superTypes[$superTypeName]);
-            }
-        }
+        $superTypes = [];
+        $this->addInheritedSuperTypes($superTypes, $this);
 
-        return array_unique($superTypes);
+        return $superTypes;
```

Comment:
This change seems dangerous to me. Can you elaborate what you did here and how it's related?

Reply:
This will skip the `null`-check from line 188ff, no?
Also, why is this change required for the feature?

Reply:
The check is also performed in `addInheritedSuperTypes()` ([L202ff](https://github.com/neos/neos-development-collection/pull/2166/files#diff-2d2cbc04188c2098aeb5811e7bdc0fe9R203)).

And yes - I did this despite it isn't required - at least I made a separate commit, so I can pretty easily remove the changes. So should I make another PR with only the feature? I don't think I will make one with the other commit then... 🤔